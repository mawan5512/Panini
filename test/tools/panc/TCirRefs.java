/*
 * This file is part of the Panini project at Iowa State University.
 *
 * The contents of this file are subject to the Mozilla Public License
 * Version 1.1 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/.
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * For more details and the latest version of this code please see
 * http://paninij.org
 *
 * Contributor(s): Sean L. Mooney
 */

/*
 * @test
 * @summary Make sure circular references are detected and warned about.
 * @compile/fail/ref=TCirRefs.out -Werror -XDrawDiagnostics TCirRefs.java
 */

capsule C(D d, E e) { void foo(){}}

capsule D(C c) { void foo(){} }

capsule E(C c) { void foo(){} }

capsule F(F f) { void foo() {} }

capsule M() {
    design {
        C c;
        D d;
        E e;
        c(d, e);
        d(c);
        e(c);
    }
}

capsule N() {
    design {
        F f1, f2;
        f1(f2);
        f2(f1);
    }
}


