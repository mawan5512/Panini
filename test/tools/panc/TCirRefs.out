- compiler.warn.cyclic.references.exists: e, c, M
- compiler.warn.cyclic.references.exists: d, c, M
- compiler.err.warnings.and.werror
1 error
2 warnings
